pkg/kubeapiserver/authenticator/config.go

```diff
@@ -148,33 +156,61 @@ func (config Config) New() (authenticator.Request, *spec.SecurityDefinitions, sp
 	// cache misses for all requests using the other. While the service account plugin
 	// simply returns an error, the OpenID Connect plugin may query the provider to
 	// update the keys, causing performance hits.
+	var updateAuthenticationConfig func(*apiserver.AuthenticationConfiguration) error
 	if config.AuthenticationConfig != nil {
-		for _, jwtAuthenticator := range config.AuthenticationConfig.JWT {
-			var oidcCAContent oidc.CAContentProvider
-			if len(jwtAuthenticator.Issuer.CertificateAuthority) > 0 {
-				var oidcCAError error
-				oidcCAContent, oidcCAError = dynamiccertificates.NewStaticCAContent("oidc-authenticator", []byte(jwtAuthenticator.Issuer.CertificateAuthority))
-				if oidcCAError != nil {
-					return nil, nil, nil, oidcCAError
+		var jwtAuthenticatorPtr atomic.Pointer[jwtAuthenticatorWithCancel]
+		var synchronousInitialization bool
+		updateAuthenticationConfig = func(authConfig *apiserver.AuthenticationConfiguration) error {
```

Comment:
I don't think re-using `updateAuthenticationConfig` with `synchronousInitialization=false` for the init on construction is helpful... literally only the `newJWTAuthenticator` line is shared. Trying to figure out the differences the first time through is confusing.

Can we make the initial construction more straightforward:

```go
initialJWTAuthenticator, err := newJWTAuthenticator(serverLifecycle, authConfig, config.OIDCSigningAlgs, config.APIAudiences, config.ServiceAccountIssuers)
if err != nil {
  return err
}
var jwtAuthenticatorPtr atomic.Pointer[jwtAuthenticatorWithCancel]
jwtAuthenticatorPtr.Load(initialJWTAuthenticator)
```

Then make a standalone updater type that holds all the state required to implement `updateAuthenticationConfig`. This isolates the reload logic from ambient stuff available to the closure and makes it super clear what the inputs are:

```go
updateAuthenticationConfig = (&authenticationConfigUpdater{
  serverLifecycle: serverLifecycle,
  config: c,
  jwtAuthenticatorPtr: jwtAuthenticatorPtr,
}).updateAuthenticationConfig

type authenticationConfigUpdater struct {
  serverLifecycle context.Context
  config Config
  jwtAuthenticatorPtr atomic.Pointer[jwtAuthenticatorWithCancel]
}
```

Finally, can the timeout for `updateAuthenticationConfig` to return be controlled by a passed-in context rather than a hard-coded-to-public package-var internal timeout?

```go
// ctx controls the timeout for updateAuthenticationConfig to return, not the lifetime of the constructed authenticators
func (u *authenticationConfigUpdater) updateAuthenticationConfig(ctx context.Context, authConfig *apiserver.AuthenticationConfiguration) error {
  ... existing logic, but assuming synchronous init
}
```